src/main/java/gyro/google/compute/DiskResource.java

```diff
@@ -0,0 +1,288 @@
+            if (error != null) {
+                throw new GyroException(error.toPrettyString());
+            }
+
+            refresh();
+        } catch (Exception ex) {
+            throw new GyroException(ex.getMessage(), ex.getCause());
+        }
+    }
+
+    @Override
+    public List<ValidationError> validateMore() {
+        return new ArrayList<>();
+    }
+
+    @Override
+    public boolean refresh() {
+        Compute client = createComputeClient();
+
+        try {
+            Disk disk = client.disks().get(getProjectId(), getZone(), getName()).execute();
+            copyFrom(disk);
+
+            return true;
+        } catch (GoogleJsonResponseException je) {
+            if (je.getDetails().getMessage().matches("The resource (.*) was not found")) {
+                return false;
+            } else {
+                throw new GyroException(je.getDetails().getMessage());
+            }
+        } catch (IOException ex) {
+            throw new GyroException(ex.getMessage(), ex.getCause());
+        }
+    }
+
+    @Override
+    public void delete(GyroUI ui, State state) {
+        Compute compute = createComputeClient();
+
+        try {
+            compute.disks().delete(getProjectId(), getZone(), getName()).execute();
+        } catch (IOException ex) {
+            throw new GyroException(ex.getMessage(), ex.getCause());
+        }
+    }
+
+    @Override
+    public void update(GyroUI ui, State state, Resource current, Set<String> changedFieldNames) {
+        Compute client = createComputeClient();
+
+        if (changedFieldNames.contains("size-gb")) {
+            saveSizeGb(client, (RegionDiskResource) current);
+        }
+
+        if (changedFieldNames.contains("labels")) {
+            saveLabels(client);
+        }
+
+        if (changedFieldNames.contains("resource-policies")) {
+            saveResourcePolicies(client, (RegionDiskResource) current);
+        }
+
+        refresh();
+    }
+
+    private void saveSizeGb(Compute client, RegionDiskResource oldRegionDiskResource) {
+        if (getSizeGb() < oldRegionDiskResource.getSizeGb()) {
+            throw new GyroException("Size of the disk cannot be decreased once set.");
+        }
+
+        try {
+            DisksResizeRequest resizeRequest = new DisksResizeRequest();
+            resizeRequest.setSizeGb(getSizeGb());
+            client.disks().resize(getProjectId(), getZone(), getName(), resizeRequest).execute();
+        } catch (IOException ex) {
+            throw new GyroException(ex.getMessage(), ex.getCause());
+        }
+    }
+
+    private void saveLabels(Compute client) {
+        try {
+            ZoneSetLabelsRequest labelsRequest = new ZoneSetLabelsRequest();
+            labelsRequest.setLabels(getLabels());
+            labelsRequest.setLabelFingerprint(getLabelFingerprint());
+            client.disks().setLabels(getProjectId(), getZone(), getName(), labelsRequest).execute();
+        } catch (IOException ex) {
```

Comment:
Should catch the `GoogleJsonResponseException`, which has better details of the error message.
```suggestion
        } catch (GoogleJsonResponseException ex) {
```